pkg/plugins/policies/meshhttproute/plugin/v1alpha1/listeners.go

```diff
@@ -32,10 +32,13 @@ func generateFromService(
 	rules rules.ToRules,
 	svc meshroute_xds.DestinationService,
 ) (*core_xds.ResourceSet, error) {
-	tags := svc.BackendRef.Tags
-	listenerBuilder := envoy_listeners.NewOutboundListenerBuilder(proxy.APIVersion, svc.OutboundInterface.DataplaneIP, svc.OutboundInterface.DataplanePort, core_xds.SocketAddressProtocolTCP).
+	var tags envoy_tags.Tags
+	if svc.Outbound.LegacyOutbound != nil {
```

Comment:
the one thing that scares me about this is that one has to remember to always check LegacyOutbound for tags. Isn't `svc.Outbound.Tags` or similar safer?

Or like `envoy_listeners.ConfigureFromOutbound(outbound xds_types.Outbound)` or `xds_types.Outbound.Configure(listener envoy_listener.Listener)` ? The tags thing is so "leaky" here.

Reply:
Updated the PR with `TagsOrNil` method